fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -514,6 +517,52 @@ public ShowResultSet visitShowTableStatement(ShowTableStmt statement, ConnectCon
             return new ShowResultSet(statement.getMetaData(), rows);
         }
 
+        @Override
+        public ShowResultSet visitShowTemporaryTablesStatement(ShowTemporaryTableStmt statement, ConnectContext context) {
+            ShowTemporaryTableStmt showTemporaryTableStmt = statement;
+            List<List<String>> rows = Lists.newArrayList();
+            String catalogName = showTemporaryTableStmt.getCatalogName();
+            if (catalogName == null) {
+                catalogName = context.getCurrentCatalog();
+            }
+            if (!CatalogMgr.isInternalCatalog(catalogName)) {
+                return new ShowResultSet(showTemporaryTableStmt.getMetaData(), rows);
```

Comment:
> This can be misleading, if we don't support tmp table in external catalog for now, should return an error to user explicitly(check this in analysis phase)

fixed, add a check in analysis phase